### DISCUSSION TITLE
Is it possible to rename depName for lookup only?

### DISCUSSION THREAD

**Discussion by kwelch** [renovatebot/renovate#25307] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.40.4

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We have an internal proxy for docker images, but it can only be used to read/download images. It does not allow for inspecting and parsing versions. 

I am hoping that I can remap the `depName` during the lookup phase to our registry so that it can find these updates, without changing the `depName` in the resulting PR. 

Example:

Input `depName`
```
docker-proxy.interal.com/sox/buildeng/agent-baseagent
```
`depName` for lookup
```
docker.internal-paas.net/sox/buildeng/agent-baseagent
```

This could be for a `docker` or `bitbucket-pipelines` managers

Current File
```
image:
  name: docker-proxy.interal.com/sox/buildeng/agent-baseagent:1.0.0
```

Ideal PR
```
image:
  name: docker-proxy.interal.com/sox/buildeng/agent-baseagent:2.0.0
```

### What you tried so far.

I have tried a package rule for this as a generic, which did not work on the replace. However during that testing I also realized that it would likely change the output and that is something I am hoping to avoid. 

```
{
      matchDatasources: ['docker', 'bitbucket-pipelines'],
      matchPackagePatterns: ['^docker-proxy.interal.com/'],
      replacementNameTemplate:
        "{{{replace 'docker-proxy.interal.com/' 'docker.internal-paas.net/' packageName}}}",
}
```

### Relevant debug logs

```
{
  "dep":
    {
      "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
      "currentValue": "1.5.7",
      "datasource": "docker",
      "depName": "docker-proxy.internal.com/sox/buildeng/agent-baseagent",
      "depType": "docker",
      "packageName": "docker-proxy.internal.com/sox/buildeng/agent-baseagent",
      "replaceString": "docker-proxy.internal.com/sox/buildeng/agent-baseagent:1.5.7",
      "updates": {},
      "versioning": "docker",
      "warnings":
        [
          {
            "message": "Failed to look up docker package docker-proxy.internal.com/sox/buildeng/agent-baseagent",
            "topic": "docker-proxy.internal.com/sox/buildeng/agent-baseagent",
          },
        ],
    },
  "env": "dev",
  "host_type": "k8s",
  "level": 20,
  "logContext": "kubernetes-bitbucket-cloud",
  "manager": "bitbucket-pipelines",
  "msg": "packageFile dep update",
  "name": "renovate",
  "packageFile": { "packageFile": "bitbucket-pipelines.yml" },
  "pid": 25,
  "region": "us-west-2",
  "renovate_version": "36.40.4",
  "repository": "test/tester",
  "time": "2023-10-19T18:53:20.521Z",
}
```

**Comment by viceice** [comment, marked as answer]
https://docs.renovatebot.com/configuration-options/#registryaliases

**Reply by kwelch** [reply]
Perfect! I didn't see pipelines in the supported list, so I patched it in to our version. I was going to PR it, but appears it has been added since 🎉 🎉 🎉

**Reply by viceice** [reply]
feel free to open a PR to update the docs 🤗
